app/models/pii/scrubber/replacement_pii.rb

```diff
@@ -0,0 +1,48 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'faker'
+
+module Pii::Scrubber
+  class ReplacementPii
+    attr_accessor :max_level
+
+    STATIC_TYPE_VALUES = {
+      first_name: 'FirstName%{id}',
+      middle_name: 'MiddleName%{id}',
+      last_name: 'LastName%{id}',
+      full_name: 'FullName%{id}',
+      ssn: '999-00-0000',
+      dob: Date.new(2000, 1, 1),
+      email: 'no-reply@example.com',
+      phone: '212-555-0100',
+      geo_street: 'Street%{id}',
+      geo_postal_code: '00000',
+    }.freeze
+
+    def self.static_value(field)
+      raise "can't scrub required field '#{field.description}" unless STATIC_TYPE_VALUES.key?(field.type)
+
+      format(TYPES_VALUES[field.type], id: record.id)
+    end
+
+    def self.fake_value(field)
+      case field.type
+      when :ssn
+        # use 999 to make it more obvious the ssn is invalid
+        Faker::IdNumber.invalid.gsub(/^\d{3}/, '999')
+      when :first_name
+        Faker::Name.first_name
+      when :last_name
+        Faker::Name.last_name
+      when :middle_name
+        Faker::Name.middle_name
+      when :full_name
+        Faker::Name.name
+      end
```

Comment:
This doesn't encompass all of the keys in `STATIC_TYPE_VALUES`.  I haven't found where we're calling `fake_value` but should this extend to cover the whole set of keys or raise on unknown key?

Reply:
Or maybe it just return `nil` which is fine for the other keys

Reply:
This is called by the FakeScrubber. Scrubbers are stacked together so if you scrub with `custom_scrubber: :fake` it fills in what it can with fake values using this method. The other fields are handled either by the dob scrubber or the basic scrubber (catch-all).

So this method doesn't have to be comprehensive but we could add in more fields if we find it useful. I didn't include emails, phones, and addresses in `fake_value` since I want to make sure they are easily distinguishable from real values.